src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,197 @@
+               {:new-network-values networks
+                :network-bars       network-bars
+                :amount->width      amount->width
+                :reset-values-fn    reset-state-values
+                :lock-press-fn      lock-press
+                :unlock-press-fn    unlock-press
+                :add-new-timeout    add-new-timeout-fn})))
+         [requesting-data?])
+        [:<>
+         (doall
+          (for [[bar-idx bar] (map-indexed vector network-bars)
+                :let          [bar-max-width      (amount->width (:max-amount bar))
+                               bar-width          (-> (:amount-shared-value bar)
+                                                      (reanimated/get-shared-value)
+                                                      (amount->width))
+                               hide-division?     (or (= last-bar-idx bar-idx) @selecting-network?)
+                               this-bar-selected? (= @selected-network-idx bar-idx)
+                               ;; Side effect: adds this bar's animation to the atom
+                               _ (add-hide-animation bar-idx
+                                                     {:amount->width amount->width
+                                                      :bar           bar})]]
+            ^{:key (str "network-bar-" bar-idx)}
+            [:f> network-bar
+             {:bar           bar
+              :max-width     bar-max-width
+              :total-width   total-width
+              :total-amount  total-amount
+              :bar-division? hide-division?
+              :on-top?       this-bar-selected?
+              :allow-press?  (and (or (not @selecting-network?) this-bar-selected?)
+                                  (not requesting-data?)
+                                  (not @press-locked?))
+              :on-press      #(when-not @selecting-network?
+                                (let [[previous-bars [_ & next-bars]] (split-at bar-idx network-bars)
+                                      number-previous-bars            bar-idx]
+                                  (animation/move-previous-bars
+                                   {:bars                 previous-bars
+                                    :bars-widths-negative bars-widths-negative})
+                                  (animation/move-pressed-bar
+                                   {:bar                  bar
+                                    :bars-widths-negative bars-widths-negative
+                                    :number-previous-bars number-previous-bars})
+                                  (animation/move-next-bars
+                                   {:bars                 next-bars
+                                    :bars-widths-negative bars-widths-negative
+                                    :number-previous-bars (inc number-previous-bars)
+                                    :extra-offset         (- bar-max-width bar-width)
```

Comment:
:thinking: I understand your worries about calculations, I added that `max` suggestion, but I think is not a good idea to wrap pieces of code in a "secure" statement so often, it will make the code harder to read and to debug IMO, particularly in this case **I think** there's no chance for it to break :thinking:

Reply:
I might be exaggerating, but I just tend to assume collections can be empty in Clojure for way too many reasons. Maybe for `networks` it's fine, the coll really won't ever be empty. Just a word of caution that designers tend to design for the happy path because the UI is very high level, they may not know `networks` could be dynamically built and for some reason could be empty.